### PR TITLE
Download task support

### LINKED ISSUE #10: Add support for download tasks
This will require refactoring a significant amount. Right now most of the recording logic lives in the data task. We'll need to make this more abstract so it can work for both.

We could also make download tasks use data tasks internally and just write it out to a temp file. The consumer won't know the difference.

### PR DESCRIPTION
**Pull request by soffes**
Fixes #10

Just using data task internally. Easy enough.

### CODE REVIEW

File: DVR/SessionDownloadTask.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class SessionDownloadTask: NSURLSessionDownloadTask {
+
+    // MARK: - Types
+
+    typealias Completion = (NSURL?, NSURLResponse?, NSError?) -> Void
```
Comment (hyperspacemark) [inline DVR/SessionDownloadTask.swift:5]: Can we yank this `typealias` up to the `Session` or something so we don't have to declare it twice?

**Comment by hyperspacemark** [comment]
One comment. Good to merge though. :green_apple:

**Comment by soffes** [comment]
They're different 😉

Sam

**Comment by hyperspacemark** [comment]
@soffes ayy lmao

**Pull request merged**